ctags/parser.go

```diff
@@ -50,8 +50,8 @@ type parseResult struct {
 
 func (lp *CTagsParser) Parse(name string, content []byte, typ CTagsParserType) ([]*Entry, error) {
 	if lp.parsers[typ] == nil {
-		parser, err := lp.newCTagsParser(typ)
-		if err != nil {
+		parser, err := lp.newParserProcess(typ)
+		if parser == nil {
```

Comment:
are we sure it isn't possible for parser to be non-nil? Maybe just in case check || err != nil?